utils/providers.py

```diff
@@ -286,6 +287,15 @@ def setup_provider(provider_key, validate=True, check_existing=True):
 
     """
     provider = get_crud(provider_key)
+    if hasattr(provider, "infra_provider"):
+        if provider.infra_provider not in {None, False}:
+            from cfme.infrastructure.provider import OpenstackInfraProvider
+            if isinstance(provider.infra_provider, OpenstackInfraProvider):
+                infra_provider_name = provider.infra_provider.name
+            else:
+                infra_provider_name = str(provider.infra_provider)
+            infra_provider_key = get_provider_key(infra_provider_name)
+            setup_provider(infra_provider_key, validate=validate, check_existing=check_existing)
```

Comment:
I don't remember why exactly was I putting it here. To be consistent with the setup_provider parameters? Not sure.